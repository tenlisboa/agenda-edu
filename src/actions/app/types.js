export const SET_EVENTS = 'SET_EVENTS'
export const SELECT_EVENT = 'SELECT_EVENT'
export const BUSY = 'BUSY'
