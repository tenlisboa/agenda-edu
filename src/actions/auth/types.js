export const ERROR = 'ERROR'
export const BUSY = 'BUSY'
