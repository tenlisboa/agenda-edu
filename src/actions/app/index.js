import api from '../../utils/api'
import { AsyncStorage } from 'react-native'
import { SET_EVENTS } from './types'

export const fetchEvents = (page = 1) => async dispatch => {
  const token = await AsyncStorage.getItem('@Auth:Token')
  const { data } = await api.get(`/events?limit=10;page=${page}`, {
    headers: { token }
  })
  dispatch({
    type: SET_EVENTS,
    events: data
  })
}
