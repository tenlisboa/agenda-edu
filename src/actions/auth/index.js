import { ERROR, BUSY } from './types'
import api from '../../utils/api'
import { AsyncStorage } from 'react-native'

export const login = (email, password) => async dispatch => {
  dispatch({ type: BUSY, busy: true })
  try {
    const { data } = await api.post('/login', { email, password })
    await AsyncStorage.setItem('@Auth:Token', data.token)
    dispatch({ type: BUSY, busy: false })
    dispatch({ type: ERROR, error: null })
    return data.token
  } catch (err) {
    dispatch({ type: ERROR, error: err.response.data.message })
    dispatch({ type: BUSY, busy: false })
  }
}
