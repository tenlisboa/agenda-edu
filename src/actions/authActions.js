import { LOGIN } from './actionTypes'

export const login = (email, password) => {
  const token = 'aksdjfaoidsfj12308rfa0fn23fna'

  return {
    type: LOGIN,
    payload: token
  }
}
