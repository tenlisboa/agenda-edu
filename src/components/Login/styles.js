import styled from 'styled-components/native'
import { Platform } from 'react-native'
import { getStatusBarHeight } from 'react-native-status-bar-height'

export const Screen = styled.View`
  margin-top: ${getStatusBarHeight()}px;
  padding-left: 34px;
  padding-right: 34px;
`

export const Container = styled.View`
  background-color: #fff;
`

export const Input = styled.TextInput`
  ${props =>
    props.inFocus ? 'border: 2px solid #733DBE;' : 'border: 1px solid #abb1b7;'}
  border-radius: 4px;
  height: 66px;
  padding-left: 26px;
  position: relative;
`

export const Title = styled.Text`
  font-size: 33px;
  font-family: 'System';
  font-weight: bold;
`

export const Label = styled.Text`
  font-size: 14px;
  color: #666666;
  font-family: 'System';
  font-weight: 300;
`
