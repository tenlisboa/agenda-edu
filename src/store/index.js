import { createStore, combineReducers, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'

import Reducers from '../reducers'

export const store = createStore(
  combineReducers(Reducers),
  applyMiddleware(thunk)
)
