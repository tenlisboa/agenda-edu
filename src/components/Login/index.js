import React, { Component } from 'react'
import { View } from 'react-native'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { login } from '../../actions/auth'

import InputText from '../Shared/InputText'

import { Screen, Container, ButtonBlock, Title, Label } from './styles'

class Login extends Component {
  state = {
    email: 'student@ae.com',
    password: '123456',
    inEmailFocus: false,
    inPasswordFocus: false,
    showPassword: false
  }

  handleEmail = email => {
    this.setState({ email })
  }

  handlePassword = password => {
    this.setState({ password })
  }

  handleEmailFocus = () => {
    this.setState({ inEmailFocus: true, inPasswordFocus: false })
  }

  handlePasswordFocus = () => {
    this.setState({ inPasswordFocus: true, inEmailFocus: false })
  }

  handleShowHidePassword = () => {
    this.setState({ showPassword: !this.state.showPassword })
  }

  submit = async () => {
    const { login } = this.props
    const { email, password } = this.state

    await login(email, password)

    if (this.props.token) {
      this.props.navigation.navigate('Events')
    }
  }

  render() {
    const { error } = this.props
    const {
      inEmailFocus,
      inPasswordFocus,
      showPassword,
      password,
      email
    } = this.state
    return (
      <Screen>
        <Container>
          <Title>FaÃ§a seu login ðŸ”‘</Title>
          <View style={{ marginTop: 47 }}>
            <Label>E-mail ou usuÃ¡rio</Label>
            <InputText
              iconName="envelope"
              iconColor="#AAAAAA"
              inFocus={inEmailFocus}
              onFocus={this.handleEmailFocus}
              value={email}
              onChangeText={email => this.handleEmail(email)}
              hasError={!!error}
              error={error}
            />
          </View>
          <View style={{ marginTop: 15 }}>
            <Label>Senha</Label>
            <InputText
              iconName={showPassword ? 'eye' : 'eye-slash'}
              iconColor="#AAAAAA"
              iconTouchable={true}
              iconOnPress={this.handleShowHidePassword}
              secureTextEntry={!showPassword}
              inFocus={inPasswordFocus}
              onFocus={this.handlePasswordFocus}
              value={password}
              onChangeText={password => this.handlePassword(password)}
            />
          </View>
        </Container>

        <ButtonBlock title="Entrar" onPress={this.submit} />
      </Screen>
    )
  }
}

const mapStateToProps = state => ({
  token: state.auth.token,
  error: state.auth.error
})

const mapDispatchToProps = dispatch => bindActionCreators({ login }, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login)
