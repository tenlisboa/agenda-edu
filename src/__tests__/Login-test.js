import React from 'react'
import { Provider } from 'react-redux'
import Login from '../components/Login'
import { store } from '../store'

import * as actions from '../actions/auth'
import * as types from '../actions/auth/types'
import reducer from '../reducers/auth/authReducer'

import renderer from 'react-test-renderer'

test('Login renders correctly', () => {
  const container = () => (
    <Provider store={store}>
      <Login />
    </Provider>
  )
  const tree = renderer.create(container()).toJSON()
  expect(tree).toMatchSnapshot()
})

test('login actions has logged successfull', async () => {
  const { username, password } = {
    username: 'student@ae.com',
    password: '123456'
  }

  expect(actions.login(username, password)).toBeTruthy()
})

test('should handle error message', () => {
  const initialState = {
    error: null,
    busy: false
  }

  expect(
    reducer(initialState, {
      type: types.ERROR,
      error: 'Invalid username or password'
    })
  ).toEqual({
    error: 'Invalid username or password',
    busy: false
  })
})
