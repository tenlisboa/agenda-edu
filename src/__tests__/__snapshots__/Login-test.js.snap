// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginTop": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "paddingLeft": 34,
          "paddingRight": 34,
          "paddingTop": "27%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "System",
              "fontSize": 33,
              "fontWeight": "bold",
            },
          ]
        }
      >
        FaÃ§a seu login ðŸ”‘
      </Text>
      <View
        style={
          Object {
            "marginTop": 47,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#666666",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "300",
                "marginBottom": 10,
              },
            ]
          }
        >
          E-mail ou usuÃ¡rio
        </Text>
        <View
          style={
            Array [
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "position": "absolute",
                "right": "6%",
                "top": 20,
                "zIndex": 1,
              }
            }
          >
            <Text />
          </View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            error={null}
            hasError={false}
            iconColor="#AAAAAA"
            iconName="envelope"
            inFocus={false}
            onChangeText={[Function]}
            onFocus={[Function]}
            style={
              Array [
                Object {
                  "borderColor": "#999999",
                  "borderRadius": 4,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": 66,
                  "paddingLeft": 26,
                  "position": "relative",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="student@ae.com"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 15,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#666666",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "300",
                "marginBottom": 10,
              },
            ]
          }
        >
          Senha
        </Text>
        <View
          style={
            Array [
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "position": "absolute",
                "right": "6%",
                "top": 20,
                "zIndex": 1,
              }
            }
          >
            <Text />
          </View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            iconColor="#AAAAAA"
            iconName="eye-slash"
            iconOnPress={[Function]}
            iconTouchable={true}
            inFocus={false}
            onChangeText={[Function]}
            onFocus={[Function]}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "borderColor": "#999999",
                  "borderRadius": 4,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": 66,
                  "paddingLeft": 26,
                  "position": "relative",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="123456"
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#733DBE",
          "borderRadius": 7,
          "bottom": 70,
          "left": 34,
          "opacity": 1,
          "paddingBottom": 21,
          "paddingTop": 21,
          "position": "absolute",
          "right": 34,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 21,
              "fontWeight": "500",
              "textAlign": "center",
            },
          ]
        }
      >
        Entrar
      </Text>
    </View>
  </View>
</View>
`;
